src/java/com/puppetlabs/ssl_utils/ExtensionsUtils.java

```diff
@@ -864,7 +891,17 @@ private static Integer getGnTagFromName(String name) {
                 if (ret.get(type) == null) {
                     ret.put(type, new ArrayList<String>());
                 }
-                String name = (String) asn1ObjToObj(generalName.getName());
+
+                String name;
+                switch (generalName.getTagNo()) {
+                    case GeneralName.iPAddress:
+                        name = octetStringToIpString((ASN1OctetString)generalName.getName());
```

Comment:
Happened across this, figured worth sharing:

<pre><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#187">187</a>  <font color="#7f0055"><b>switch</b></font> (tag)<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#188">188</a>  {<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#189">189</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0otherName" title="int otherName" class="hidden">otherName</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#190">190</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, ASN1Sequence.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/ASN1Sequence.java#ASN1Sequence.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.ASN1Sequence.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : ASN1Sequence">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#191">191</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0rfc822Name" title="int rfc822Name" class="hidden">rfc822Name</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#192">192</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, DERIA5String.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/DERIA5String.java#DERIA5String.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.DERIA5String.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : DERIA5String">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#193">193</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0dNSName" title="int dNSName" class="hidden">dNSName</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#194">194</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, DERIA5String.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/DERIA5String.java#DERIA5String.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.DERIA5String.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : DERIA5String">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#195">195</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0x400Address" title="int x400Address" class="hidden">x400Address</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#196">196</a>      <font color="#7f0055"><b>throw</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b27/java/lang/IllegalArgumentException.java#IllegalArgumentException" title="java.lang.IllegalArgumentException">IllegalArgumentException</a>(<font color="#2a00ff">"unknown tag: "</font> + tag);<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#197">197</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0directoryName" title="int directoryName" class="hidden">directoryName</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#198">198</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, X500Name.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x500/X500Name.java#X500Name.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.x500.X500Name.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : X500Name">getInstance</a>(tagObj, <font color="#7f0055"><b>true</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#199">199</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0ediPartyName" title="int ediPartyName" class="hidden">ediPartyName</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#200">200</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, ASN1Sequence.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/ASN1Sequence.java#ASN1Sequence.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.ASN1Sequence.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : ASN1Sequence">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#201">201</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0uniformResourceIdentifier" title="int uniformResourceIdentifier" class="hidden">uniformResourceIdentifier</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#202">202</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, DERIA5String.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/DERIA5String.java#DERIA5String.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.DERIA5String.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : DERIA5String">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#203">203</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0iPAddress" title="int iPAddress" class="hidden">iPAddress</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#204">204</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, ASN1OctetString.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/ASN1OctetString.java#ASN1OctetString.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.ASN1OctetString.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : ASN1OctetString">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#205">205</a>  <font color="#7f0055"><b>case</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName.0registeredID" title="int registeredID" class="hidden">registeredID</a>:<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#206">206</a>      <font color="#7f0055"><b>return</b></font> <font color="#7f0055"><b>new</b></font> <a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#GeneralName" title="org.bouncycastle.asn1.x509.GeneralName">GeneralName</a>(tag, ASN1ObjectIdentifier.<a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/ASN1ObjectIdentifier.java#ASN1ObjectIdentifier.getInstance%28org.bouncycastle.asn1.ASN1TaggedObject%2Cboolean%29" title="org.bouncycastle.asn1.ASN1ObjectIdentifier.getInstance(org.bouncycastle.asn1.ASN1TaggedObject,boolean) : ASN1ObjectIdentifier">getInstance</a>(tagObj, <font color="#7f0055"><b>false</b></font>));<br/><a href="http://grepcode.com/file/repo1.maven.org/maven2/org.bouncycastle/bcprov-jdk15on/1.52/org/bouncycastle/asn1/x509/GeneralName.java#207">207</a>  }<br/></pre>

We can get a sense of the expected data types; `GeneralName#getName` returns an `ASN1Encodable` but when it's an IP address it's an octet string